txp/Pos/Txp/Toil/Types.hs

```diff
@@ -64,7 +77,21 @@ utxoF :: Format r (Utxo -> r)
 utxoF = later formatUtxo
 
 -- | Wrapper for genesis utxo.
-newtype GenesisUtxo = GenesisUtxo { unGenesisUtxo :: Utxo }
+newtype GenesisUtxo = GenesisUtxo
+    { unGenesisUtxo :: Utxo
+    }
+
+data GenesisTxpContext = UnsafeGenesisTxpContext
+    { _gtcUtxo         :: !GenesisUtxo
+    , _gtcStakeholders :: !GenesisStakeholders
+    }
+
+makeLenses ''GenesisTxpContext
+
+mkGenesisTxpContext :: Utxo -> GenesisTxpContext
+mkGenesisTxpContext utxo = UnsafeGenesisTxpContext
+    (GenesisUtxo utxo)
+    (GenesisStakeholders . getKeys . utxoToStakes $ utxo)
```

Comment:
Actually it's inefficient, but I don't mind because it should be done only on start.